Source/WebKit/WebProcess/WebPage/WebPage.cpp

```diff
@@ -3454,6 +3454,14 @@ void WebPage::didDismissContextMenu()
 
 #endif // ENABLE(CONTEXT_MENUS)
 
+void WebPage::showContextMenuFromFrame(const WebCore::FrameIdentifier& frameID, const ContextMenuContextData& contextMenuContextData, const UserData& userData)
+{
+    flushPendingEditorStateUpdate();
+    m_contextMenuInvoked = true;
```

Comment:
If we set this bit before calling `flush`, could we avoid `scheduleFullEditorStateUpdate` below?

Reply:
Nope

Reply:
Alright, can we just put `m_hasEverDisplayedContextMenu = true;` on the line before `scheduleFullEditorStateUpdate()` then?